src/views/Export/index.tsx

```diff
@@ -0,0 +1,380 @@
+                type: notify.type.ERROR,
+                message: error,
+                duration: notify.duration.MEDIUM,
+            });
+        },
+    });
+
+    const sanitizedFilters = useMemo(() => {
+        const processedFilters = getFiltersForRequest(leadsFilters);
+        // Unprotected filter is sent to request to fetch leads
+        // if user cannot create export for confidential documents
+        if (filterOnlyUnprotected) {
+            processedFilters.confidentiality = ['unprotected'];
+        }
+        return processedFilters;
+    }, [filterOnlyUnprotected, leadsFilters]);
+
+    const [
+        leadsPending,
+        ,
+        ,
+        getLeads,
+    ] = useRequest<MultiResponse<Lead>>({
+        url: 'server://v2/leads/filter/',
+        method: 'POST',
+        query: {
+            fields: ['id', 'title', 'created_at'],
+        },
+        body: {
+            project: [projectId],
+            ...sanitizedFilters,
+        },
+        onSuccess: (response) => {
+            const newLeads: SelectedLead[] = [];
+            (response.results || []).forEach((l) => {
+                newLeads.push({
+                    selected: true,
+                    ...l,
+                });
+            });
+            setLeads(newLeads);
+        },
+        onFailure: (error) => {
+            notify.send({
+                title: _ts('export', 'leadsLabel'),
+                type: notify.type.ERROR,
+                message: error,
+                duration: notify.duration.MEDIUM,
+            });
+        },
+    });
+
+    const [
+        geoOptionsPending,
+        ,
+        ,
+        getGeoOptions,
+    ] = useRequest<unknown>({
+        url: 'server://geo-options/',
+        method: 'GET',
+        query: {
+            project: projectId,
+        },
+        schemaName: 'geoOptions',
+        onSuccess: (response) => {
+            setGeoOptions({ projectId, locations: response });
+        },
+        onFailure: (error) => {
+            notify.send({
+                title: _ts('export', 'geoLabel'),
+                type: notify.type.ERROR,
+                message: error,
+                duration: notify.duration.MEDIUM,
+            });
+        },
+    });
+
+    useEffect(() => {
+        getAnalysisFramework();
+        getGeoOptions();
+    },
+    [getAnalysisFramework, getGeoOptions, projectId]);
+
+    useEffect(
+        getLeads,
+        [projectId, leadsFilters],
+    );
+
+    useEffect(() => {
+        setActiveExportTypeKey('word');
+        setPreviewId(undefined);
+        setReportStructure([]);
+        setDecoupledEntries(true);
+    }, [projectId]);
+
+    useEffect(() => {
+        const textWidgetList = getTextWidgetsFromFramework(analysisFramework);
+        const contextualWidgetList = getContextualWidgetsFromFramework(analysisFramework);
+        setTextWidgets(textWidgetList);
+        setContextualWidgets(contextualWidgetList);
```

Comment:
Couldn't these be replaced by a memo, and we don't really need to set the text and contextual widgets in state?

Reply:
the textwidget and contextualwidget selection is done by the child component. how are we going to set them if we don't use state?

Reply:
What @AdityaKhatri means to say is that textWidgetList and contextualWidgetList can be calculated using useMemo and doesn't need a useEffect to watch change over analysisFramework.